novawallet/Modules/PayCard/PayCardPresenter.swift

```diff
@@ -1,16 +1,75 @@
 import Foundation
+import SoraFoundation
 
 final class PayCardPresenter {
     weak var view: PayCardViewProtocol?
     let wireframe: PayCardWireframeProtocol
     let interactor: PayCardInteractorInputProtocol
 
+    private let localizationManager: LocalizationManagerProtocol
+
+    private var openCardTimestamp: TimeInterval?
+    private var cardStatus: PayCardStatus?
+
     init(
         interactor: PayCardInteractorInputProtocol,
-        wireframe: PayCardWireframeProtocol
+        wireframe: PayCardWireframeProtocol,
+        localizationManager: LocalizationManagerProtocol
     ) {
         self.interactor = interactor
         self.wireframe = wireframe
+        self.localizationManager = localizationManager
+    }
+
+    private func showFailAlert() {
+        let languages = localizationManager.selectedLocale.rLanguages
+
+        let title = R.string.localizable.cardOpenFailedAlertTitle(preferredLanguages: languages)
+        let message = R.string.localizable.cardOpenFailedAlertMessage(preferredLanguages: languages)
+
+        let closeAction = AlertPresentableAction(
+            title: R.string.localizable.commonOk(preferredLanguages: languages),
+            handler: { [weak self] in
+                self?.wireframe.close(from: self?.view)
```

Comment:
Why do we close the widget? Probably user want's to retry or get any information from the state